vip-api/src/main/java/fr/insalyon/creatis/vip/api/KeycloakSpringSecurityConfig.java

```diff
@@ -122,6 +122,7 @@ protected void configure(HttpSecurity http) throws Exception {
             .authorizeRequests()
                 .antMatchers("/rest/platform").permitAll()
                 .antMatchers("/rest/authenticate").permitAll()
+                .antMatchers("/rest/executions/{executionId}/summary").permitAll() // We can secure it by IP, but it's changed because of reverse proxy
```

Comment:
Why not authorize the shanoir IP like for the `/rest/register` URL ?

Reply:
the IP of the shanoir microservice machine is either hidden or changed because of Nginx reverse proxy as described in the description and the comment, 
I tested the IP whitelisting. when I do the curl within the machine (with the whitelisted IP) it works, but behind an Nginx reverse proxy, it doesn't.  

I think we can fix this if we call VIPs URL `http://vip-url/rest/executions/summary` directly from the shanoir microservice instand of the reversed proxy URL `https://shanoir-ng-front/shanoir-ng/vip/rest/executions/summary`
But is this a good thing to do?

What do you think?

Reply:
At first glance, I don't see any objection to call VIP from the internal URL, as it should stay in a secure network.
I prefer that to a public URL giving access to some possibly sensitive data.
I wanted to explore the keycloak features to have a token-secured communication between Shanoir and VIP but have not found the time to do it.